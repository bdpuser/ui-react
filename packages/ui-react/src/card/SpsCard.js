import React, { Component } from "react";
import classnames from "classnames";

class SpsCardHeader extends Component {
  render() {
    const { header } = this.props;
    const cardHeader =
      typeof header === "function"
        ? header
        : () => {
            return <h4 className="sps-card__title">{header}</h4>;
          };
    return (
      <div className={`sps-card__header`}>{ cardHeader()}</div>
    );
  }
}

export class SpsCard extends Component {
  render() {
    const { children, header, footer, className, ...rest } = this.props;
    const cardClasses = classnames("sps-card", className);
    const cardBodyClasses = classnames("sps-card__body", {
      "has-header": header
    });
    const cardFooter = typeof footer === "function" ? footer : () => footer;
    return (
      <div className={cardClasses}>
        {header ? (
          <SpsCardHeader header={header} />
        ) : null}
        <div className={cardBodyClasses} {...rest}>
          {children}
        </div>
        {footer ? (
          <div className={`sps-card__footer`}>{cardFooter()}</div>
        ) : null}
      </div>
    );
  }
}

export default SpsCard;
