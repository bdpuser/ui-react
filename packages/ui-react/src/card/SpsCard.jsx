import React, { Component } from "react";
import classnames from "classnames";

export class SpsCard extends Component {
  render() {
    const { children, header, footer } = this.props;
    const cardHeader = typeof header === "function" ? header : () => header;
    const cardClass = classnames("sps-card__body", { "has-header": header });
    const cardFooter = typeof footer === "function" ? footer : () => footer;

    return (
      <div className={`sps-card`}>
        {header ? (
          <div className={`sps-card__header`}>{cardHeader()}</div>
        ) : null}
        <div className={cardClass}>{children}</div>
        {footer ? (
          <div className={`sps-card__footer`}>{cardFooter()}</div>
        ) : null}
      </div>
    );
  }
}

export default SpsCard;