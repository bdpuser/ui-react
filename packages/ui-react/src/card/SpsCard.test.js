import React, { Fragment } from "react";
import { SpsCard } from "./SpsCard";
import { mount } from "enzyme";

fdescribe("SpsCard", () => {
  let mountedCard;
  let props;
  const mountCardComponent = () => {
    return mount(<SpsCard {...props}>{props.children}</SpsCard>);
  };
  afterEach(() => {
    props = {};
    mountedCard = null;
  });
  describe("SpsCard with header", () => {
    it("should display a text header", () => {
      props = {
        header: "I'm a header",
        children: "hello"
      };
      mountedCard = mountCardComponent();
      expect(mountedCard.find(".sps-card__header").length).toEqual(1);
      expect(mountedCard.find(".sps-card__title").text()).toEqual(
        "I'm a header"
      );
    });
    it("should display custom html within a header", () => {
      let renderHeader = () => {
        return (
          <Fragment>
            <i className="sps-icon sps-icon-file" />
            <span>Custom Header</span>
          </Fragment>
        );
      };
      props = {
        header: renderHeader,
        children: "hello"
      };
      mountedCard = mountCardComponent();
      expect(mountedCard.find(".sps-card__header").length).toEqual(1);
      expect(
        mountedCard
          .find(".sps-card__title")
          .containsMatchingElement(renderHeader)
      ).toEqual(true);
    });
  });
  describe("SpsCard default", () => {
    it("should display body content", () => {
      props = {
        children: "This is the card body text"
      };
      mountedCard = mountCardComponent();
      expect(mountedCard.find(".sps-card__body").text()).toEqual(
        "This is the card body text"
      );
    });
  });
  describe("SpsCard with footer", () => {
    it("should display a text footer", () => {
      props = {
        children: "This should have a text footer.",
        footer: "Here is a text footer"
      };
      mountedCard = mountCardComponent();
      expect(mountedCard.find(".sps-card__footer").text()).toEqual(
        "Here is a text footer"
      );
    });
    it("should display custom html within a footer", () => {
      let renderFooder = () => {
        return <div>Here is a custom Footer</div>;
      };
      props = {
        footer: renderFooder,
        children: "This should have a custom footer"
      };
      mountedCard = mountCardComponent();
      expect(mountedCard.find(".sps-card__footer").length).toEqual(1);
      expect(
        mountedCard
          .find(".sps-card__footer")
          .containsMatchingElement(renderFooder)
      ).toEqual(true);
    });
  });
});
