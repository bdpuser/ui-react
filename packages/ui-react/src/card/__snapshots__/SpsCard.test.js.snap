// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpsCard SpsCard snapshot renders correctly 1`] = `
<div
  className="sps-card"
>
  <div
    className="sps-card__header"
  >
    <h4
      className="sps-card__title"
    >
      I'm a header
    </h4>
  </div>
  <div
    className="sps-card__body has-header"
  >
    hello
  </div>
</div>
`;
